enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+```
+# HELP cluster_installation_invocation represents a single command run by the OpenShift installer and related outcomes.
+# TYPE cluster_installation_invocation counter
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="25"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="22.5"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="APIFailure",return="1",version="4.2",duration="30"} 1
+cluster_installation_invocation{command="destroy", target="cluster",os="linux",platform="AWS",result="Success",return="1",version="4.2",duration="10"} 1
+```
+* `duration` of the command represnts the overall time required to complete the command. The label values will be bounded in the installer client to preset buckets of a certain step-size, 2.5 in the example above.
+* `return` - denotes success or failure. It could potentially be subsumed in result.
+* `result` label denotes a command outcome:
+    - `ProvisioningFailed` - for `create` the installer failed to provision resources
+    - `BootstrapFailed` - for `create` the bootstrap node failed
+    - `APIFailed` - for `create` and `wait-for` the control plane API never came up
+    - `OperatorsDegraded` - for `create` and `wait-for` command not all operators became available
+    - `Success` - for `create` and `wait-for` the cluster completed successfully and other commands completed successfully
+    - TODO: determine relevant results for other commands, perhaps inability to connect to SSH with `destroy` or `gather`
+
+#### cluster_installation_duration
+`cluster_installation_duration` tracks the duration of each stage of installation in the execution of the `create` and `wait-for` commands. Similar to `duration` in the previous stage, the value would be bound to certain buckets with a step size:
+
+```
+# HELP cluster_installation_duration keeps track of the count of the duration of individual stages of the command execution in buckets with the result of the command.
+# TYPE cluster_installation_duration counter
+cluster_installation_duration{infrastructure="5", bootstrap="5", api="5", operators="5" success="true"} 1
+```
+All label values would be rounded up to the nearest step size, so if it took 4 minutes to complete bootstrapping and the agreed upon step size was 2.5 minutes, the value would be rounded to 5.
+
+The proposed periods to track are:
+* `infrastructure`
+* `bootstrap`
+* `api`
+* `operators`
+
+One sample would be pushed per command invocation.
+
+#### cluster_installation_modification
+The `cluster_installation_modification` metric represents whether a user modified an installer created asset prior to running the `create` command.
```

Comment:
I think instead of having per file modification,

1) we track if modification was made
2) we define categories, like machines, machineconfigs, scheduler, network, ingress, dns so that we are tracking use-cases and not noise.